adlfs/transfer.py

```diff
@@ -256,18 +253,17 @@ def submit(self, src, dst, length):
             'pending', 'running', 'finished', 'cancelled', 'errored')
 
         # Create unique temporary directory for each file
-        if self._tmp_unique and self._tmp_path:
-            tmpdir = os.path.join(self._tmp_path, uuid.uuid4().hex)
+        if self._chunked and self._unique_temporary:
+            tmpdir = dst.parent / "{}.segments.{}".format(dst.name, self._unique_str)
```

Comment:
Agree with @martindurant here. For download the directory will need to be created if it doesn't exist first, right? For upload this is not required since our service will create the entire path to the file automatically.